internal/newrelic/client_test.go

```diff
@@ -189,3 +189,82 @@ func TestNrClient_FindEntityByGUID(t *testing.T) {
 		})
 	}
 }
+
+func Test_checkBounds(t *testing.T) {
+	lowerResult := 5.0
+	upperResult := 15.0
+
+	type args struct {
+		actualResult        any
+		expectedLowerResult *float64
+		expectedUpperResult *float64
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "when the actual result is below of a dual-bound it should return an error",
+			args:    args{actualResult: 2.0, expectedLowerResult: &lowerResult, expectedUpperResult: &upperResult},
+			wantErr: true,
+		},
+		{
+			name:    "when the actual result is above of a dual-bound it should return an error",
+			args:    args{actualResult: 20.0, expectedLowerResult: &lowerResult, expectedUpperResult: &upperResult},
+			wantErr: true,
+		},
+		{
+			name:    "when the actual result is below of an lower bound it should return an error",
+			args:    args{actualResult: 1.0, expectedLowerResult: &lowerResult, expectedUpperResult: nil},
+			wantErr: true,
+		},
+		{
+			name:    "when the actual result is above of an upper bound it should return an error",
+			args:    args{actualResult: 16.0, expectedLowerResult: nil, expectedUpperResult: &upperResult},
+			wantErr: true,
+		},
+		{
+			name:    "when the actual result is above of an lower bound it should return no error",
+			args:    args{actualResult: 6.0, expectedLowerResult: &lowerResult, expectedUpperResult: nil},
+			wantErr: false,
+		},
+		{
+			name:    "when the actual result is equal to a lower bound it should return no error",
+			args:    args{actualResult: 5.0, expectedLowerResult: &lowerResult, expectedUpperResult: nil},
+			wantErr: false,
+		},
+		{
+			name:    "when the actual result is below of an upper bound it should return no error",
+			args:    args{actualResult: 14.0, expectedLowerResult: nil, expectedUpperResult: &upperResult},
+			wantErr: false,
+		},
+		{
+			name:    "when the actual result is equal to an upper bound it should return no error",
+			args:    args{actualResult: 15.0, expectedLowerResult: nil, expectedUpperResult: &upperResult},
+			wantErr: false,
+		},
+		{
+			name:    "when the actual result is between a dual-bound it should return no error",
+			args:    args{actualResult: 10.0, expectedLowerResult: &lowerResult, expectedUpperResult: &upperResult},
+			wantErr: false,
+		},
+		{
+			name:    "when the actual result is nil an error should be thrown",
+			args:    args{actualResult: nil, expectedLowerResult: &lowerResult, expectedUpperResult: &upperResult},
+			wantErr: true,
+		},
+		{
+			name:    "when bounds are missing an error should be thrown",
+			args:    args{actualResult: 99.0, expectedLowerResult: nil, expectedUpperResult: nil},
+			wantErr: true,
+		},
```

Comment:
Could we also include test cases for different types? For instance, should `actualResult: "10.0"` work?

Reply:
This would not work since bounded comparisons only support ints/floats.